lib/index.js

```diff
@@ -53,13 +53,53 @@ function applyMethods(q) {
     });
   };
 
-  q.series = function(arr, body) {
+  q.series = function (arr, body) {
     return arr.reduce(function(promise, item) {
       return promise.then(function(result) {
         return body(item, result);
       });
     }, q());
   };
+
+  /**
+   * Allows body up to timeout ms to complete, and retries it if exceeds that time.
+   * Body will not be retried if it fails for any other reason (e.g. exception or rejection).
+   *
+   * @param {Object} options
+   * @param {number} options.maxRetries - Max number of times to retry body.
+   * @param {Object} options.cancel - An object which will have a cancel method created on, allowing the retry loop to be canceled early.
+   */
+  q.timeoutRetry = function (timeout, body, options) {
+    options = options || {};
+
+    var timeoutIdentifier = 'timeout-dc1836808a';
+    var maxRetries = options.maxRetries || 5;
+    var attempts = 0;
+    var success = false;
+    var canceled = false;
+
+    if (options.cancel) {
+      options.cancel.cancel = function () {
+        canceled = true;
+      }
+    }
+
+    return q.while(function () {
+      return !success;
+    }, function () {
+      attempts += 1;
+      return q.fcall(body).timeout(timeout, timeoutIdentifier).tap(function () {
+        success = true;
+      }).catch(function (err) {
+        if (canceled)
+          throw new Error('Canceled');
+        if (err.message === timeoutIdentifier && attempts === maxRetries)
```

Comment:
That was just my automatic linter.
Then, I looked a little deeper & figured, since these were numbers, the `==` (or `===`) was a touch dangerous.